lib/travis/model/job.rb

```diff
@@ -122,13 +122,8 @@ def decrypted_config
     normalize_config(self.config).deep_dup.tap do |config|
       config[:env] = process_env(config[:env]) { |env| decrypt_env(env) } if config[:env]
       config[:global_env] = process_env(config[:global_env]) { |env| decrypt_env(env) } if config[:global_env]
-      if config[:addons]
-        if addons_enabled?
-          config[:addons] = decrypt_addons(config[:addons])
-        else
-          delete_addons(config)
-        end
-      end
+      delete_addons(config) if config[:addons] && !addons_enabled?
+      config[:addons] = decrypt_addons(config[:addons]) if config[:addons]
```

Comment:
@BanzaiMan thoughts? Do you think this is good enough to merge? We can go ahead to modify `delete-addons` if you think that's the only way to go, but would prefer to avoid that if you think there are other options.